File: app/src/main/java/org/skyfaced/mvp/service/network/WaifuRepository.java
package org.skyfaced.mvp.service.network;

import org.skyfaced.mvp.model.ImageDto;
import org.skyfaced.mvp.util.WaifuType;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

public interface WaifuRepository {
    Observable<ImageDto> waifu(WaifuType type);

    Observable<List<ImageDto>> waifus(WaifuType type);
}

File: app/src/main/java/org/skyfaced/mvp/service/network/WaifuRepository.kt
package org.skyfaced.mvp.service.network

import io.reactivex.rxjava3.core.Observable
import org.skyfaced.mvp.model.ImageDto
import org.skyfaced.mvp.util.WaifuType

interface WaifuRepository {
    fun waifu(type: WaifuType): Observable<ImageDto>

    fun waifus(type: WaifuType): Observable<List<ImageDto>>
}